client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx

```diff
@@ -1,53 +1,108 @@
-
+import { initiateThemeTransfer } from 'calypso/state/themes/actions';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
 /**
  * Style dependencies
  */
 import 'calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status/style.scss';
+import { getPurchaseFlowState } from 'calypso/state/plugins/marketplace/selectors';
 
 const StyledProgressBar = styled( ProgressBar )`
 	margin: 20px 0px;
 `;
 
-function WrappedMarketplacePluginSetup( {
-	pluginSlug = 'wordpress-seo',
-}: {
-	pluginSlug?: string;
-} ): JSX.Element {
+function WrappedMarketplacePluginSetup(): JSX.Element {
+	const [ currentStage, setCurrentStage ] = useState( 1 );
+	const [ simulatedProgressPercentage, setSimulatedProgressPercentage ] = useState( 0 );
+	const stages = [ 'Installing plugin', 'Activating plugin' ];
+	const interval = 2000;
+
+	const dispatch = useDispatch();
 	const selectedSite = useSelector( getSelectedSite );
+	const transferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, selectedSite?.ID )
+	);
+	const { pluginSlugToBeInstalled, isPluginInstalledDuringPurchase } = useSelector(
+		getPurchaseFlowState
+	);
+
+	useEffect( () => {
+		if ( pluginSlugToBeInstalled ) {
+			dispatch( initiateThemeTransfer( selectedSite.ID, null, pluginSlugToBeInstalled ) );
+		} else if ( isPluginInstalledDuringPurchase ) {
+			alert( 'To be implemented: polling wait to be implemented to check for transfer' );
+		} else {
+			// Invalid State redirect to Yoast marketplace page for now, and maybe a marketplace home view in the future
+			page(
+				`/marketplace/product/details/wordpress-seo${
+					selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+				}`
+			);
+		}
+	}, [ dispatch, pluginSlugToBeInstalled ] );
+
+	useEffect( () => {
+		setTimeout(
+			() => {
+				if ( simulatedProgressPercentage < 95 ) {
+					setSimulatedProgressPercentage( ( percentage ) => percentage + 6 );
+				} else if ( simulatedProgressPercentage !== 100 && simulatedProgressPercentage !== 95 ) {
+					setSimulatedProgressPercentage( 95 );
+				}
+			},
+
+			interval
+		);
+	}, [ simulatedProgressPercentage ] );
 
-	//Plugin status can be extracted with this selector once installation is triggered
-	const pluginStatus = useSelector( ( state ) => {
-		return selectedSite ? getStatusForPlugin( state, selectedSite?.ID, pluginSlug ) : 'UNKNOWN';
-	} );
+	useEffect( () => {
+		if ( transferStatus === 'complete' ) {
+			// TODO: Make sure the primary domain is set to the relevant domain before redireting to thank-you page
+			setSimulatedProgressPercentage( 100 );
+			setTimeout( () => {
+				page(
+					`/marketplace/thank-you${
+						selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+					}?flags=marketplace-yoast`
+				);
+			}, 4000 );
+		}
+	}, [ selectedSite.slug, transferStatus ] );
 
+	if ( simulatedProgressPercentage > 50 && currentStage === 1 ) {
```

Comment:
Let's define constants for the stage index number so it's more readable, instead of just 1 and 2 in the following line :)

Reply:
Fixed by e9fdeddd6ff5993e0b6f277657512c58fec957a9